File: sqliteDBViewModelDemo/app/src/main/java/edu/cs4730/sqlitedbviewmodeldemo/MainActivity.java
package edu.cs4730.sqlitedbviewmodeldemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import edu.cs4730.sqlitedbviewmodeldemo.databinding.ActivityMainBinding;
import edu.cs4730.sqlitedbviewmodeldemo.db.CursorViewModel;

import android.database.Cursor;
import android.os.Bundle;
import android.view.View;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {
    ActivityMainBinding binding;
    myAdapter mAdapter;

    CursorViewModel mViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        mViewModel = new ViewModelProvider(this).get(CursorViewModel.class);

        binding.list.setLayoutManager(new LinearLayoutManager(this));
        binding.list.setItemAnimator(new DefaultItemAnimator());
        mAdapter = new myAdapter(null,getApplicationContext());
        //add the adapter to the recyclerview
        binding.list.setAdapter(mAdapter);

        mViewModel.getData().observe(this, new Observer<Cursor>() {
            @Override
            public void onChanged(Cursor data) {
                mAdapter.setCursor(data);
            }
        });

        binding.floatingActionButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mViewModel.add("Jim", 3012);
                mViewModel.add("Danny", 312);
            }
        });
    }
}

File: sqliteDBViewModelDemo_kt/app/src/main/java/edu/cs4730/sqlitedbviewmodeldemo_kt/MainActivity.kt
package edu.cs4730.sqlitedbviewmodeldemo_kt

import android.database.Cursor
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import edu.cs4730.sqlitedbviewmodeldemo_kt.databinding.ActivityMainBinding
import edu.cs4730.sqlitedbviewmodeldemo_kt.db.CursorViewModel

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    lateinit var mAdapter: myAdapter
    lateinit var mViewModel: CursorViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)
        ViewCompat.setOnApplyWindowInsetsListener(binding.main) { v: View, insets: WindowInsetsCompat ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            WindowInsetsCompat.CONSUMED
        }
        mViewModel = ViewModelProvider(this).get(CursorViewModel::class.java)

        binding.list.layoutManager = LinearLayoutManager(this)
        binding.list.itemAnimator = DefaultItemAnimator()
        mAdapter = myAdapter(null, applicationContext)
        //add the adapter to the recyclerview
        binding.list.adapter = mAdapter
        mViewModel.data.observe(this, Observer<Cursor?> { data -> mAdapter.setCursor(data) })

        binding.floatingActionButton.setOnClickListener {
            mViewModel.add("Jim", 3012)
            mViewModel.add("Danny", 312)
        }
    }
}